compose/service.py

```diff
@@ -514,9 +514,9 @@ def _next_container_number(self, one_off=False):
 
     def _get_aliases(self, container):
         if container.labels.get(LABEL_ONE_OFF) == "True":
-            return []
+            return set()
 
-        return [self.name, container.short_id]
+        return set([self.name, container.short_id])
```

Comment:
minor: the `[]` are unnecessary here, you can actually do use a set literal now that we dropped python 2.6 support:

```python
return {self.name, container.short_id}
```